seacrowd/sea_datasets/burapha_th/burapha_th.py

```diff
@@ -0,0 +1,168 @@
+}
+
+_SUPPORTED_TASKS = [Tasks.IMAGE_CAPTIONING]
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+_SUBSETS = ["character", "digit", "syllable"]
+
+
+def config_constructor(subset: str, schema: str, version: str) -> SEACrowdConfig:
+    return SEACrowdConfig(
+        name=f"{_DATASETNAME}_{subset}_{schema}",
+        version=version,
+        description=f"{_DATASETNAME} {subset} {schema} schema",
+        schema=f"{schema}",
+        subset_id=f"{_DATASETNAME}_{subset}",
+    )
+
+
+class BuraphaThDataset(datasets.GeneratorBasedBuilder):
+    """
+    The dataset has 68 character classes, 10 digit classes, and 320 syllable classes.
+    For constructing the dataset, 1072 Thai native speakers wrote on collection datasheets
+    that were then digitized using a 300 dpi scanner.
+    De-skewing, detection box and segmentation algorithms were applied to the raw scans for
+    image extraction. The dataset, unlike all other known Thai handwriting datasets, retains
+    existing noise, the white background, and all artifacts generated by scanning.
+    """
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [config_constructor(subset, "source", _SOURCE_VERSION) for subset in _SUBSETS]
+    BUILDER_CONFIGS.extend([config_constructor(subset, "seacrowd_imtext", _SEACROWD_VERSION) for subset in _SUBSETS])
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_digit_source"
+
+    label_chr_dig = [str(i).zfill(2) for i in range(78)]
+    label_syl = [str(i).zfill(3) for i in range(320)]
+
+    def _info(self) -> datasets.DatasetInfo:
+        task = self.config.subset_id.split("_")[2]
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {"id": datasets.Value("string"), "image_paths": datasets.Value("string"), "label": datasets.Sequence(datasets.ClassLabel(names=self.label_chr_dig if task == "character" or task == "digit" else self.label_syl))}
+            )
+        elif self.config.schema == "seacrowd_imtext":
+            features = schemas.image_text_features(label_names=self.label_chr_dig if task == "character" or task == "digit" else self.label_syl)
+        else:
+            raise NotImplementedError()
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        task = self.config.subset_id.split("_")[2]
+
+        train_path = dl_manager.download_and_extract(_URLS[task]["train"])
+        test_path = dl_manager.download_and_extract(_URLS[task]["test"])
```

Comment:
I'd suggest to change so that the `dl_manager.download_and_extract` is being called once rather than twice:
```suggestion
        _local_path = dl_manager.download_and_extract(_URLS[task]) 
        train_path, test_path = _local_path["train"], _local_path["test"]
```

Reply:
Oh we can do that?
Cool!